tests/tests_lite/conftest.py

```diff
@@ -1,7 +1,104 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 import os
 from typing import List
 
 import pytest
+import torch.distributed
+
+
+@pytest.fixture(scope="function", autouse=True)
+def preserve_global_rank_variable():
```

Comment:
Do you think we will be able to remove some of these fixtures from the PL conftest?

Reply:
I see these more as safety nets. I would be very careful removing them. We haven't seen any big maintenance effort around them so I wouldn't worry about it atm. I want to copy them over for now so that I can later add test without losing much time on debugging.